src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,26 @@
+import { StringParam, useQueryParams } from 'use-query-params';
+
+type Props = {
+  sortBy: string,
+};
+
+export const Icon: React.FC<Props> = ({ sortBy }) => {
+  const [searchParams] = useQueryParams({
+    sort: StringParam,
+    order: StringParam,
+  });
+
+  const isSorting = searchParams.sort === sortBy;
+  const isDesc = searchParams.order === 'desc';
+
+  switch (true) {
```

Comment:
Avoid using switch(true), it can lead to unexpected results. It can be replaced with simpler and more readable if-else statements.